### ISSUE TITLE
[Errno 86] Bad CPU type in executable: 'Users/abcd/Library/Application Support/undetected_chromedriver/....._chromedriver

### ISSUE THREAD

**Issue by a-maxime** [ultrafunkamsterdam/undetected-chromedriver#1020]
Hi there,

It seems that I'm having trouble having the good chromedriver for my CPU architecture. Long story short. I run effortlessly undetected-chromedriver on my Mac M1 14' (with M1 Pro) but I tried to do the same on another and I keep getting the same error again and again :

`[Errno 86] Bad CPU type in executable: 'Users/abcd/Library/Application Support/undetected_chromedriver/....._chromedriver`

I tried to download myself the Mac arm64 version of chromedriver from Google website and call `driver = uc.Chrome(driver_executable_path= "some path")` and then I got another error `selenium.common.exceptions.WebDriverException: Message: Service /Users/emmacuchet/Documents/Shopmium/Automatisation/chromedriver unexpectedly exited. Status code was: -9`

**My MBP M1 14' :**
selenium : 4.7.2
undetected-chromedriver : 3.1.7
Chrome v109.0.5414.119 (arm64)
bash : bin/zsh
Mac OS Monterey 12.5.1 

**The second MBP M1 14' (the one having trouble) :**
selenium : tried 4.7.2 & 4.8.0
undetected-chromedriver : tried 3.1.7 & 3.2.0 & 3.2.1
Chrome v109.0.5414.119 (arm64)
bash : bin/zsh
Mac OS Ventura 13.1

Thanks a lot,
Max

**Comment by a-maxime** [comment]
Solution found : I had to install Rosetta 2 with the command `softwareupdate --install-rosetta`

That's means with uc or bs4 that Python is trying to use the x86 package :/

**Comment by ultrafunkamsterdam** [comment]
Weird as M1 uses arm64 iiirc . Good to hear it solved

**Comment by a-maxime** [comment]
> Weird as M1 uses arm64 iiirc . Good to hear it solved

Yes really weird, it seems that it uses Rosetta2 to run Chrome. Don’t really know why I have this behavior. Anyway, it’s solved but not a « normal » way of working
